rust-sdk/whirlpool/src/decrease_liquidity.rs

```diff
@@ -27,20 +27,96 @@ use crate::{
 
 // TODO: support transfer hooks
 
+/// Represents the parameters for decreasing liquidity in a pool.
+///
+/// You must specify only one of the parameters (`TokenA`, `TokenB`, or `Liquidity`).
+/// Based on the provided value, the SDK computes the other two parameters.
+///
+/// # Variants
```

Comment:
Here you can put the rustdoc comment next to the Variant itself I think and it will then show up nicely in the rustodc